src/legacy/core_plugins/kibana/public/dashboard/dashboard_state_manager.ts

```diff
@@ -450,60 +504,62 @@ export class DashboardStateManager {
    *
    * @returns {DashboardViewMode}
    */
-  getViewMode() {
+  public getViewMode() {
     return this.hideWriteControls ? DashboardViewMode.VIEW : this.appState.viewMode;
   }
 
   /**
    * @returns {boolean}
    */
-  getIsViewMode() {
+  public getIsViewMode() {
     return this.getViewMode() === DashboardViewMode.VIEW;
   }
 
   /**
    * @returns {boolean}
    */
-  getIsEditMode() {
+  public getIsEditMode() {
     return this.getViewMode() === DashboardViewMode.EDIT;
   }
 
   /**
    *
    * @returns {boolean} True if the dashboard has changed since the last save (or, is new).
    */
-  getIsDirty(timeFilter) {
+  public getIsDirty(timeFilter?: TimeRange) {
```

Comment:
is this supposed to be `Timefilter`? Or is `TimeRange` correct?

Reply:
ah, I think Timefilter.  Unfortunately because `Timefilter` is typed as `any`, I can pass anything along and it doesn't complain.  I'll see what kind of rabbit hole I get into if I can give Timefilter the most basic shape so it would have caught this.